src/inventory.cpp

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2020 YADRO
+
+#include "inventory.hpp"
+
+#include "config.hpp"
+
+#include <sdbusplus/bus.hpp>
+
+/**
+ * @brief Construct item name from its path.
+ *
+ * @param[in] path - D-Bus path of the item
+ *
+ * @return inventory item name
+ */
+static std::string nameFromPath(const std::string& path)
+{
+    size_t lastSlash = path.find_last_of('/');
+
+    // some objects (such as CPU's cores) have non-unique name,
+    // for these items we should include parent node as prefix
+    if (lastSlash && lastSlash != std::string::npos)
+    {
+        // check for core item
+        static const std::string coreName = "core";
+        if (path.compare(lastSlash + 1, coreName.length(), coreName) == 0)
+        {
+            // include CPU name (parent node)
+            lastSlash = path.find_last_of('/', lastSlash - 1);
+        }
+    }
+
+    if (lastSlash == std::string::npos)
+        return path;
+    else
+        return path.substr(lastSlash + 1);
+}
+
+/**
+ * @brief Comparator for human sorting of inventory items.
+ *
+ * Comparsion based on digits inside the name:
+ * cpu1/core2 is less than cpu1/core10, but grater than cpu0/core10.
+ *
+ * @param[in] a - first item to compare
+ * @param[in] b - second item to compare
+ *
+ * @return comparsion result, true if a < b
+ */
+static bool humanCompare(const InventoryItem& a, const InventoryItem& b)
+{
+    static const char* digits = "0123456789";
+
+    size_t lastPos = 0;
+    while (lastPos < a.name.length())
```

Comment:
We should not recalculate the length of the name on each iteration.
I suggest to define a constant outside of the loop.